src/conn/conn_dhandle.c

```diff
@@ -575,14 +575,23 @@ __wt_conn_btree_apply(WT_SESSION_IMPL *session, const char *uri,
             WT_WITH_HANDLE_LIST_READ_LOCK(
               session, WT_DHANDLE_NEXT(session, dhandle, &conn->dhqh, q));
             if (dhandle == NULL)
-                return (0);
+                break;
+
+            if (WT_IS_LAS(dhandle)) {
+                las_dhandle = dhandle;
+                continue;
+            }
 
             if (!F_ISSET(dhandle, WT_DHANDLE_OPEN) || F_ISSET(dhandle, WT_DHANDLE_DEAD) ||
               dhandle->type != WT_DHANDLE_TYPE_BTREE || dhandle->checkpoint != NULL ||
               WT_IS_METADATA(dhandle))
                 continue;
             WT_ERR(__conn_btree_apply_internal(session, dhandle, file_func, name_func, cfg));
         }
+        /* Move the LAS dhandle to the end for checkpointing. */
+        if (las_dhandle != NULL)
```

Comment:
Don't we need to add this part (LAS to the end of dhandle list) incase of user-given checkpoint target list?

Reply:
I guess I was of the opinion that if the user specifies targets then we shouldn't add in extra things.